web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -371,6 +371,8 @@
               scope.gnCurrentEdit.associatedPanelConfigId = attrs['configId'] || 'default';
               scope.relations = {};
               scope.gnCurrentEdit.codelistFilter  = attrs['codelistFilter'];
+              scope.isMdWorkflowEnableForMetadata = gnConfig['metadata.workflow.enable'] &&
+                angular.isDefined(scope.gnCurrentEdit.metadata.mdStatus);
```

Comment:
If it is fine for approved metadata then the check should be mdStatus === 2 ?

Reply:
indeed. so maybe we could just disable it if the current record is a draft ?

Reply:
Sounds fine, I'll update the code.

Reply:
@fxprunayre updated the code. Doi can be published for a resource if:

  - Doi publication is enabled.
  - The resource matches doi.org url
  - The workflow is not enabled for the metadata and the metadata is published.